### PR TITLE
feat: add yaml format target

### PR DESCRIPTION
**Pull request by viktorvoltaire**
Since we are adding yaml as a dependency in https://github.com/einride/mage-tools/pull/61

### CODE REVIEW

File: targets/mgyamlfmt/targets.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"go.einride.tech/mage-tools/mgtool"
+	"gopkg.in/yaml.v3"
+)
+
+func FormatYaml() error {
+	return filepath.Walk(mgtool.GetGitRootPath("."), func(path string, _ os.FileInfo, err error) error {
```
Comment (viktorvoltaire) [inline targets/mgyamlfmt/targets.go:13]: Should we support excluding directories ? Right now im just defaulting to the git root

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1,13 +1,12 @@
 version: 2
-
```
Comment (viktorvoltaire) [inline .github/dependabot.yml:2]: Not a 100% sure about this format tbh, feels like blank lines should be kept?

Comment (viktorvoltaire) [reply .github/dependabot.yml:2]: Haven’t looked into https://pkg.go.dev/gopkg.in/yaml.v3#Style might be something there

Comment (viktorvoltaire) [reply .github/dependabot.yml:2]: Seems like there isnt an option for this

**Comment by viktorvoltaire** [comment]
I pushed a new commit to preserve blank lines. It does the job, in a somewhat hacky manner.

File: .github/dependabot.yml (hunk 2 above)
Comment (viktorvoltaire) [reply .github/dependabot.yml]: See my latest commit for a potential solution

File: targets/mgyamlfmt/targets.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	"go.einride.tech/mage-tools/mglog"
+	"go.einride.tech/mage-tools/mgtool"
+	"gopkg.in/yaml.v3"
+)
+
+func FormatYaml() error {
+	logger := mglog.Logger("format-yaml")
+	logger.Info("formatting yaml files...")
+	return filepath.Walk(mgtool.GetGitRootPath("."), func(path string, _ fs.FileInfo, err error) error {
```
Comment (Edholm) [inline targets/mgyamlfmt/targets.go:19]: btw I think [`filepath.WalkDir`](https://pkg.go.dev/path/filepath#WalkDir) is the recommended func nowadays due to it now having to call `stat` on _every_ file. No need to change, just a heads-up

**Review by Edholm — APPROVED** [review]
Noice work!

File: .github/dependabot.yml (hunk 2 above)
Comment (odsod) [reply .github/dependabot.yml]: :star_struck:

**Comment by odsod** [comment]
Files ending in multiple blank lines might be able to be handled by doing `bytes.TrimSpace` and then appending a single trailing endline.

**Comment by viktorvoltaire** [comment]
> Files ending in multiple blank lines might be able to be handled by doing `bytes.TrimSpace` and then appending a single trailing endline.

Ah true!

File: targets/mgyamlfmt/targets.go (hunk 3 above)
Comment (viktorvoltaire) [reply targets/mgyamlfmt/targets.go]: Fixed

File: targets/mgyamlfmt/targets.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	}
+	var b bytes.Buffer
+	encoder := yaml.NewEncoder(&b)
+	encoder.SetIndent(2)
+	if err := encoder.Encode(&node); err != nil {
+		return err
+	}
+	// Remove temporary comment.
+	indentPreserveComment := regexp.MustCompile("\n\\s+#preserveEmptyLine\n")
+	out := b.String()
```
Comment (odsod) [inline targets/mgyamlfmt/targets.go:48]: @viktorvoltaire Minor thought: The conversion to string and back to bytes here should be avoidable. (Converting back and forth allocates memory for a copy).

Comment (viktorvoltaire) [reply targets/mgyamlfmt/targets.go:48]: Ah, will have a look and see if I can fix that 👍

Comment (viktorvoltaire) [reply targets/mgyamlfmt/targets.go:48]: @odsod actually I already fixed that. Was annoyed with it 😊 https://github.com/einride/mage-tools/commit/ee6ea005bfd2721fd1165f58320a72b09ec3fe0c#diff-bd1f7610004a598ff323f80c6917c7ede5455328026fc0397bc78b7554a93d74R38

Comment (odsod) [reply targets/mgyamlfmt/targets.go:48]: Sweet

**Pull request merged**